### PR TITLE
Add laplace Functional API

### LINKED ISSUE #15742: laplace

### PR DESCRIPTION
**Pull request by trgiangdo**
Close #15742

In this PR, the functional API on paddlepaddle is commented out because there is no Laplace in paddle 2.4.2
But there will be in the next version (hopefully) since the class does exist in the develop branch.

### CODE REVIEW

File: ivy/functional/backends/jax/experimental/random.py [hunk 1]
```diff
@@ -116,3 +116,29 @@ def bernoulli(
...
+    seed: Optional[int] = None,
+    out: Optional[JaxArray] = None,
+) -> JaxArray:
+    if size is None:
+        if isinstance(loc, float) and isinstance(scale, float):
+            size = 1
+        else:
+            size = jax.numpy.broadcast_shapes(loc.shape, scale.shape)
+
+    jax.numpy.broadcast_shapes(loc.shape, scale.shape)
```
Comment (MahmoudAshraf97) [inline ivy/functional/backends/jax/experimental/random.py:138]:
```suggestion
```

File: ivy/data_classes/array/experimental/random.py [hunk 2]
```diff
@@ -271,3 +271,46 @@ def bernoulli(
...
+        *,
+        size: Optional[Union[ivy.Shape, ivy.NativeShape]] = None,
+        device: Optional[Union[ivy.Device, ivy.NativeDevice]] = None,
+        dtype: Optional[Union[ivy.Dtype, ivy.NativeDtype]] = None,
+        seed: Optional[int] = None,
+        out: Optional[ivy.Array] = None,
+    ) -> ivy.Array:
+        """
+        Parameters
+        ----------
```
Comment (MahmoudAshraf97) [inline ivy/data_classes/array/experimental/random.py:289]: please add the instance method prefix to the docstring

File: ivy/functional/backends/mxnet/experimental/random.py
Comment (MahmoudAshraf97) [inline ivy/functional/backends/mxnet/experimental/random.py:1]: you need to implement the mxnet function

File: ivy/functional/backends/paddle/experimental/random.py
Comment (MahmoudAshraf97) [inline ivy/functional/backends/paddle/experimental/random.py:1]: you need to implement the paddle function, as we support multiple versions of the frameworks so you can use a compositional implementation for now

**Review by MahmoudAshraf97 — CHANGES_REQUESTED** [review]
Hello, I have several comments please address them, in addition to those, try to use the functional API in each framework instead of creating a distribution and sampling from it

File: ivy/functional/backends/mxnet/experimental/random.py
Comment (trgiangdo) [reply ivy/functional/backends/mxnet/experimental/random.py:1]: I'm trying to implement this and test it, but it might take a while since installing `mxnet` on Mac M1 is such a pain 🥲

However, I see that no other distribution is implemented with MXNet. So I'm not sure if it may cause problems in the future.

File: ivy/functional/backends/paddle/experimental/random.py
Comment (trgiangdo) [reply ivy/functional/backends/paddle/experimental/random.py:1]: Hi, as stated in [this page](https://unify.ai/docs/ivy/overview/deep_dive/function_types.html#compositional-functions), a compositional implementation requires the function to be able to use other ivy API.

However, I'm not sure how to sample from a Laplace distribution from other ivy API (from other distribution). I will try my best.

**Comment by trgiangdo** [comment]
Hello @MahmoudAshraf97, thank you for your quick review. I have answer to your comments and trying to fix the problems.

Also,
> in addition to those, try to use the functional API in each framework instead of creating a distribution and sampling from it

I don't understand what you mean by *"use the functional API in each framework"*, since sampling from a distribution is the usage in most backend doc (TF, Paddle, Torch), or I may miss something. Other experimental distributions are also using the same method.
Can you help me clarify this point?

File: ivy/functional/backends/mxnet/experimental/random.py
Comment (trgiangdo) [reply ivy/functional/backends/mxnet/experimental/random.py:1]: `mx.np.random.laplace` will be supported with the release of mxnet2.0. Also, the `loc` and `scale` is only supported as float, not an array.

So I guess the best solution for now is a compositional API as well. But again, I'm not sure if this is possible as a random function.

**Comment by MahmoudAshraf97** [comment]
Hey @trgiangdo , How are thing going?

**Comment by trgiangdo** [comment]
I implemented laplace as compositional function. I think it is ready for you to review.

File: ivy/functional/ivy/experimental/random.py [hunk 3]
```diff
@@ -302,3 +303,107 @@ def bernoulli(
...
+    eps = _get_eps(loc.dtype)
+
+    if seed is not None:
+        ivy.seed(seed_value=seed)
+
+    if size is None:
+        size = (1,)
+
+    uniform = ivy.random_uniform(
+        low=float(np.nextafter(-1, 1)) + eps / 2,
```
Comment (MahmoudAshraf97) [inline ivy/functional/ivy/experimental/random.py:378]: we can't use np functions in ivy implementations, only ivy functions

**Review by MahmoudAshraf97 — CHANGES_REQUESTED** [review]
I extremelly apologize for the delay in review

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.
